File: app/src/main/java/com/gerardbradshaw/mixup/di/moreapps/MoreAppsComponent.java
package com.gerardbradshaw.mixup.di.moreapps;

import com.gerardbradshaw.mixup.models.AppInfo;

import javax.inject.Named;

import dagger.BindsInstance;
import dagger.Component;

@Component//(modules = MoreAppsModule.class)
public interface MoreAppsComponent {

  AppInfo getAppInfo();

  @Component.Builder
  interface Builder {

    @BindsInstance
    Builder titleRes(@Named("titleRes") int titleRes);

    @BindsInstance
    Builder descriptionRes(@Named("descriptionRes") int descriptionRes);

    @BindsInstance
    Builder iconRes(@Named("iconRes") int iconRes);

    @BindsInstance
    Builder urlRes(@Named("urlRes") int urlRes);

    MoreAppsComponent build();
  }
}

File: app/src/main/java/com/gerardbradshaw/mixup/di/moreapps/MoreAppsComponent.kt
package com.gerardbradshaw.mixup.di.moreapps

import android.content.Context
import com.gerardbradshaw.mixup.models.AppInfo
import com.gerardbradshaw.mixup.ui.moreapps.MoreAppsFragment
import dagger.BindsInstance
import dagger.Subcomponent
import javax.inject.Named

@Subcomponent(modules = [AppInfoModule::class])
interface MoreAppsComponent {

  @Subcomponent.Factory
  interface Factory {
    fun create(@BindsInstance activityContext: Context): MoreAppsComponent
  }

  //@Named("mater_info")
  //fun provideAppInfo(): AppInfo

  fun inject(fragment: MoreAppsFragment)
}